src/Wumpus.Net.Rest/Requests/Messages/ModifyMessageParams.cs

```diff
@@ -18,7 +18,7 @@ public void Validate()
         {
             if (!Content.IsSpecified || Content.Value is null)
                 Content = (Utf8String)"";
-            if (Embed.IsSpecified && Embed.Value != null)
+            if (!Embed.IsSpecified && Content == (Utf8String)"")
```

Comment:
The `Content == ""` check will already be done in `Preconditions.NotNullOrWhitespace`.
This if should probably be `if (!Embed.IsSpecified || Embed.Value == null)` to make sure content is only checked when no embed is passed.